codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -155,9 +155,6 @@ public int guess() {
     private final ChannelHandler inboundStreamHandler;
 
     private int initialOutboundStreamWindow = Http2CodecUtil.DEFAULT_WINDOW_SIZE;
-    // TODO: We may be able to optimize when we really need to call flush(...) during channelReadComplete(...)
```

Comment:
do we have a followup issue to investigate this, or you feel it is no longer necessary?

Reply:
no longer necessary.

Reply:
Can you explain why? Is it because you are now convinced that calling flush is cheap enough if there is nothing to flush (despite the pipeline activity, and pipelines being deeper than in the "low level API")?

Reply:
Yes exactly... (+ its the same we do in the Http2ConnectionHandler).

Reply:
> (+ its the same we do in the Http2ConnectionHandler).

The pipeline is generally assumed to be more shallow because it is not very extensible. We can tweak later if necessary though (just wondering if perf analysis was done).

Reply:
yeah we can followup if needed... that said I dont think so for now